### PR TITLE
Fix default environments in guide

### LINKED ISSUE #46841: Rails.logger.level for production in docs is wrong
Hi, was surprised by this line which has already existed for 8 years. Am I doing something wrong or docs are lying?
The default log level for `production` is `info`. For `development` and `test` is `debug`. You can check this yourself.

https://edgeguides.rubyonrails.org/debugging_rails_applications.html#log-levels
https://github.com/rails/rails/blame/afbde03801104e4d56dac736043f47f141a91d3e/guides/source/debugging_rails_applications.md?plain=1#L146

```ruby
(base) ➜  foo git:(main) ✗ RAILS_ENV=production ./bin/rails c
Loading production environment (Rails 7.0.4)
irb(main):001:0> Rails.logger.level
=> 1
irb(main):002:0> 
(base) ➜  foo git:(main) ✗ ./bin/rails c               
Loading development environment (Rails 7.0.4)
irb(main):001:0> Rails.logger.level
=> 0
irb(main):002:0> 
(base) ➜  foo git:(main) ✗ RAILS_ENV=test ./bin/rails c      
Loading test environment (Rails 7.0.4)
irb(main):001:0> Rails.logger.level
=> 0
```

Cheers

### PR DESCRIPTION
**Pull request by woto**
refs: 229fd2a02f
closes: #46841

### CODE REVIEW

File: guides/source/debugging_rails_applications.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ Rails.logger.level = 0 # at any time
 
 This is useful when you want to log under development or staging without flooding your production log with unnecessary information.
 
-TIP: The default Rails log level is `debug` in all environments.
+TIP: The default Rails log level is `debug` for `development` and `test` enviroments. And `info` for `production` environment.
```
Comment (sampatbadhe) [inline guides/source/debugging_rails_applications.md:146]:
```suggestion
TIP: The default Rails log level is `debug` for the `development` and `test` environments. And `info` for the `production` environment.
```

File: guides/source/debugging_rails_applications.md [hunk 2]
```diff
@@ -143,7 +143,7 @@ Rails.logger.level = 0 # at any time
 
 This is useful when you want to log under development or staging without flooding your production log with unnecessary information.
 
-TIP: The default Rails log level is `debug` in all environments.
+TIP: The default Rails log level is `debug` for `development` and `test` environments. And `info` for the `production` environment.
```
Comment (jonathanhefner) [inline guides/source/debugging_rails_applications.md:146]: To be more precise, the default is always `:debug`, but it is overridden [in `config/environments/production.rb`](https://github.com/rails/rails/blob/663e8e1ccd049dc005924f7033ae407d2d25ba8a/railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt#L61).  So what about something like:

```suggestion
TIP: The default Rails log level is `:debug`. However, it is set to `:info` for the `production` environment in the default generated `config/environments/production.rb`.
```

**Review by jonathanhefner — COMMENTED** [review]
Thank you for addressing this!

When the changes are finalized, would you mind [squashing the commits](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits)?

**Comment by jonathanhefner** [comment]
Thank you, @woto! :tada:

**Pull request merged**